### PR TITLE
refactor(trie): use `micro-bmark` for benchmarks

### PR DESCRIPTION
**Pull request by faustbrian**
For a smaller dependency footprint you could use https://github.com/paulmillr/micro-bmark and https://github.com/bestiejs/benchmark.js is also unmaintained for good so might be time to drop it anyway. Mostly a suggestion so can close it right away if you don't want it 😅

```
❯ npm run benchmarks

> @ethereumjs/trie@5.0.0-beta.3 benchmarks
> node -r ts-node/register --max-old-space-size=8024 benchmarks

-------
Benchmarking
1k-3-32-ran x 4 ops/sec @ 232ms/op
1k-5-32-ran x 4 ops/sec @ 201ms/op
1k-9-32-ran x 5 ops/sec @ 197ms/op
1k-1k-32-ran x 4 ops/sec @ 203ms/op
1k-1k-32-mir x 4 ops/sec @ 218ms/op
Checkpointing: 100 iterations x 6,378 ops/sec @ 156μs/op ± 11.16% (min: 79μs, max: 702μs)
Checkpointing: 500 iterations x 5,583 ops/sec @ 179μs/op ± 5.18% (min: 117μs, max: 1ms)
Checkpointing: 1000 iterations x 5,398 ops/sec @ 185μs/op ± 2.73% (min: 122μs, max: 928μs)
Checkpointing: 5000 iterations x 5,119 ops/sec @ 195μs/op
RAM: rss=355.3mb heap=209.7mb used=185.4mb ext=42.6mb arr=41.4mb
```

### CODE REVIEW

File: packages/trie/package.json [hunk 1]
```diff
@@ -60,14 +60,14 @@
...
     "0x": "^4.9.1",
-    "benchmark": "^2.1.4",
     "eslint": "^8.0.0",
     "karma": "^6.3.2",
     "karma-chrome-launcher": "^3.1.0",
     "karma-firefox-launcher": "^2.1.0",
     "karma-tap": "^4.2.0",
     "karma-typescript": "^5.5.3",
     "lmdb": "^2.5.3",
+    "micro-bmark": "^0.1.3",
```
Comment (paulmillr) [inline packages/trie/package.json:70]: It would set the exact version here: `0.1.3` to improve supply chain security even more

**Comment by paulmillr** [comment]
Context: benchmark@2.1.4 is [100.5KB minified](https://bundlephobia.com/package/benchmark@2.1.4); micro-bmark@0.1.3 is [1.3KB](https://bundlephobia.com/package/micro-bmark@0.1.3)

**Comment by paulmillr** [comment]
@faustbrian 0.2.0 is out

File: packages/trie/benchmarks/checkpointing.ts [hunk 2]
```diff
@@ -1,21 +1,10010 @@
-import { pseudoRandomBytes } from 'crypto'
```
Comment (paulmillr) [inline packages/trie/benchmarks/checkpointing.ts:1]: Same functionality in 6 lines:

```ts
const { keccak256 } = require('ethereum-cryptography/keccak'); // or whatever you're using here
let curr = keccak256(new Uint8Array(32));
const hexKeys = [];
for (let i = 0; i < 10000; curr = keccak256(curr), i++) {
  hexKeys.push(Buffer.from(curr).toString('hex'));
}
```

**Review by paulmillr — DISMISSED** [review]

**Comment by holgerd77** [comment]
Rebased this via UI.

**Review by holgerd77 — APPROVED** [review]
LGTM

**Pull request merged**